lib/base/defer.hpp

```diff
@@ -30,7 +30,11 @@ class Defer
 	inline
 	~Defer()
 	{
-		m_Func();
+		try {
+			m_Func();
+		} catch (...) {
+			// https://stackoverflow.com/questions/130117/throwing-exceptions-out-of-a-destructor
+		}
```

Comment:
Please split this into a separate PR.

Reply:
The first commit will disappear automatically once the PR linked above has been merged.